### PR TITLE
Tpetra: Beginning deprecation of files that start with Kokkos_

### PR DESCRIPTION
**Pull request by csiefer2**
Issue: #11459
Tests: Yes
Stakeholder feedback: n/a

There's still more to do (for a later PR), but I figured I'd at least get ye olde AT running on this to see what happens.

### CODE REVIEW

File: packages/tpetra/core/compat/CMakeLists.txt [hunk 1]
```diff
@@ -12,9 +12,18 @@ TRIBITS_CONFIGURE_FILE(${PACKAGE_NAME}_config.h)
 SET(HEADERS "")
 SET(SOURCES "")
 SET_AND_INC_DIRS(DIR ${CMAKE_CURRENT_SOURCE_DIR})
-APPEND_GLOB(HEADERS ${DIR}/*.h)
-APPEND_GLOB(HEADERS ${DIR}/*.hpp)
-APPEND_GLOB(SOURCES ${DIR}/*.cpp)
+
+
+IF (NOT ${PACKAGE_NAME}_ENABLE_DEPRECATED_CODE)
```
Comment (jhux2) [inline packages/tpetra/core/compat/CMakeLists.txt:17]: Is this logic backwards?

Comment (jhux2) [reply packages/tpetra/core/compat/CMakeLists.txt:17]: Ok, new comments clarify the situation.

**Review by jhux2 — APPROVED** [review]
lgtm

File: packages/tpetra/core/compat/CMakeLists.txt (hunk 1 above)
Comment (csiefer2) [reply packages/tpetra/core/compat/CMakeLists.txt:17]: Thanks for the feedback!

**Review by jhux2 — APPROVED** [review]
sure

**Pull request merged**